backend/src/data/sessionsDataAccess.ts

```diff
@@ -0,0 +1,53 @@
+import { pool } from '../config/db'
+import { v4 as uuidv4 } from 'uuid';
+import { ISession } from '../types/session';
+import { ISessionAccessCode } from '../types/sessionAccessCode';
+
+export const createSession = async (): Promise<ISession | null> => {
+    try {
+   
+  
+      const query = `
+        INSERT INTO sessions (id)
+        VALUES ($1) RETURNING id, active, match_found, created_at;
+      `;
+  
+      const data = [
+        uuidv4()
+      ];
+  
+      const result = await pool.query(query, data);
+  
+      return result.rows[0] || null;
+    } catch (error) {
+      console.log('Error creating session:', error);
+      throw 'Error creating session.';
+    }
+  };
+
+
+  export const createSessionAccessCode = async (session: ISession): Promise<ISessionAccessCode | null> => {
+    try {
+
+      const sessionId = session.id
+     
+      const query = `
+        INSERT INTO session_codes (id, session_id, access_code)
+        VALUES ($1, $2, $3) RETURNING id, active, access_code, session_id, created_at;
+      `;
+  
+      const data = [
+        uuidv4(),
+        sessionId,
+        uuidv4(),
```

Comment:
The code should be a short code that can be easily shared with someone, so probably an uuid is not the most suitable. You can look for other ways of generating short unique ids. The type and length should be decided with  @grazirs the UI of the client repends on the format of the code.

Reply:
@andymcbee @grazirs did you decide the length of the code. A long code as UUID can be difficult for a end user to share

Reply:
Hi, yes we did something like this: UJz-768445. Our logic is that it's big enough to have many combinations, while still being small enough to read on a device screen. 

There's a service now that generates the access token, and checks if it's unique before returning it. That value is now used instead of a uuid.

The service:
https://github.com/grazirs/movie-matcher/pull/1/files#diff-ba8d07273409c144e4554ce76018f12edb968804b863857d18956be5f0650499